src/main/java/org/alliancegenome/curation_api/services/validation/AlleleValidator.java

```diff
@@ -621,14 +619,37 @@ private List<AlleleFunctionalImpactSlotAnnotation> validateAlleleFunctionalImpac
 		return validatedFunctionalImpacts;
 	}
 
+	private boolean containsAssociation(List<AlleleGeneAssociation> associations, Long id) {
+		for (AlleleGeneAssociation association : associations) {
+			if(association.getId() == null) continue;
+			if(association.getId().equals(id)) return true; 
+		}
+		return false;
+	}
+
 	private List<AlleleGeneAssociation> validateAlleleGeneAssociations(Allele uiEntity, Allele dbEntity) {
 		String field = "alleleGeneAssociations";
 
+		List<AlleleGeneAssociation> uiAssociations = uiEntity.getAlleleGeneAssociations();
+		List<AlleleGeneAssociation> dbAssociations = dbEntity.getAlleleGeneAssociations();
+
+		Set<AlleleGeneAssociation> associationsToDelete = new HashSet<>();
+
+		if (CollectionUtils.isNotEmpty(dbAssociations)) {
+			for (AlleleGeneAssociation dbAssociation : dbAssociations) {
+				Long dbAssociationId = dbAssociation.getId();
+				if (!containsAssociation(uiAssociations, dbAssociationId)) {
```

Comment:
So instead of looping the whole set again and again in the containsAssociation it would be better to create a set and use that for lookup.

Reply:
Do you mean like `uiAssociations.contains(dbAssociation)`?